### ISSUE TITLE
Please add a typescript definition file

### ISSUE THREAD

**Issue by taheri24** [turtle-DB/turtleDB#7]
This would make it much easier to use your api with full intellisence.

**Comment by kevinclarkadstech** [comment]
Yes these days to truly be a good 3rd party JavaScript library, there needs to be TypeScript definitions.  TypeScript is self documenting so it is easier to consume a library without constantly referring to documentation or using dev tools or console to figure out what an object is or a function should return.  I think the turtledb seems like an awesome project, but will try pouchdb or rolling my own sync with something like Dexie.js which has TypeScript support.  Thanks will keep an eye on this lib! 👋
